symengine/ruby/ext/symengine/symengine_macros.h

```diff
@@ -0,0 +1,22 @@
+#ifndef SYMENGINE_MACROS_H_
+#define SYMENGINE_MACROS_H_
+
+#include "ruby.h"
+#include "symengine/cwrapper.h"
+
+VALUE Klass_of_Basic(const basic_struct *basic_ptr);
+
+#define GET_SYMINTFROMVAL(num_value, this) \
+    switch(TYPE(num_value)) { \
+        case T_FIXNUM: \
+            integer_set_si(this, FIX2LONG(num_value)); \
+            break; \
+        case T_BIGNUM: \
+            integer_set_ui(this, NUM2ULONG(num_value)); \
```

Comment:
BigNum is bigger than a long most of the time. You should get the string representation and then construct a `SymEngine::Integer`